### PR TITLE
[CN-571] Restructure AWS discovery documentation

### PR DESCRIPTION
**Pull request by SeriyBg**

### CODE REVIEW

File: docs/modules/deploy/pages/deploying-on-aws.adoc [hunk 1]
```diff
@@ -92,9 +92,108 @@ The following optional properties can be configured[[ec2_properties]]:
 
 Note that if you don't specify any of the properties, then Hazelcast uses the IAM Role assigned to the EC2 Instance to form a cluster from all Hazelcast members running in same region.
 
-=== EC2 Hazelcast Client Configuration
+=== Client Discovery of EC2 Hazelcast Cluster
 
-Hazelcast Client discovery parameters are the same as mentioned above.
+Hazelcast Client can automatically discover the Hazelcast cluster running in EC2 environment.
+
+- *Client running in EC2:*
```
Comment (hasancelik) [inline docs/modules/deploy/pages/deploying-on-aws.adoc:99]: Is it possible to fix these titles?
Screenshot 2023-07-17 at 16 58 48

File: docs/modules/deploy/pages/deploying-on-aws.adoc [hunk 2]
```diff
@@ -92,9 +92,108 @@ The following optional properties can be configured[[ec2_properties]]:
 
 Note that if you don't specify any of the properties, then Hazelcast uses the IAM Role assigned to the EC2 Instance to form a cluster from all Hazelcast members running in same region.
 
-=== EC2 Hazelcast Client Configuration
+=== Client Discovery of EC2 Hazelcast Cluster
```
Comment (hasancelik) [inline docs/modules/deploy/pages/deploying-on-aws.adoc:95]: Is it possible to list all topologies at the side listing?

Screenshot 2023-07-17 at 17 01 20

**Review by hasancelik — CHANGES_REQUESTED** [review]

File: docs/modules/deploy/pages/deploying-on-aws.adoc (hunk 2 above)
Comment (SeriyBg) [reply docs/modules/deploy/pages/deploying-on-aws.adoc:95]: As far as I can understand, only L2 titles (`==`) are listed there

File: docs/modules/deploy/pages/deploying-on-aws.adoc (hunk 1 above)
Comment (hasancelik) [reply docs/modules/deploy/pages/deploying-on-aws.adoc:99]: > What do you mean?

As you understand already, converting them into proper titles 🙁 

>  I've tried making them L4 titles (====), but they are not distinguishable from the text, then

hmm, maybe the doc team can help us with it. @Serdaro do you have any solution?

File: docs/modules/deploy/pages/deploying-on-aws.adoc (hunk 2 above)
Comment (hasancelik) [reply docs/modules/deploy/pages/deploying-on-aws.adoc:95]: I see, @Serdaro can we change this rule for this specific page?

File: docs/modules/deploy/pages/deploying-on-aws.adoc (hunk 1 above)
Comment (Serdaro) [reply docs/modules/deploy/pages/deploying-on-aws.adoc:99]: I have been using Serge's solution, too (making these supposed-to-be-headings bold, provided that they don't need to be linkables). Let me work with the UI and see.

**Review by Serdaro — COMMENTED** [review]
As a general note, I am seeing "client" mentions, instead of "clients". I was wondering if this is because we specifically talk about the Java client.

File: docs/modules/deploy/pages/deploying-on-aws.adoc (hunk 2 above)
Comment (hasancelik) [reply docs/modules/deploy/pages/deploying-on-aws.adoc:95]: Do you have any idea why AWS page does not list them?

**Review by cagric0 — APPROVED** [review]
I have tried to test some of combination of the different fields configured. But as you know it's time consuming, for now I stop doing it. And also we have separate task for adding test cases with different member and client configurations. So I am approving the PR and we may update this page later with the outcome of [this task](https://hazelcast.atlassian.net/jira/software/projects/CN/boards/42/backlog?selectedIssue=CN-907).

**Pull request merged**